completion.go

```diff
@@ -36,21 +39,22 @@ func sendResourceResponse(rw http.ResponseWriter, res []string) {
 	}
 }
 
-type tableRequest struct {
-	Schema string `json:"schema"`
-}
-
-type columnRequest struct {
-	Table string `json:"table"`
-}
-
 func (ds *sqldatasource) getSchemas(rw http.ResponseWriter, req *http.Request) {
 	if ds.Completable == nil {
 		handleError(rw, ErrorNotImplemented)
 		return
 	}
 
-	res, err := ds.Completable.Schemas(req.Context())
+	options := Options{}
+	if req.Body != nil {
+		err := json.NewDecoder(req.Body).Decode(&options)
+		if err != nil {
+			handleError(rw, err)
+			return
+		}
```

Comment:
```suggestion
		if err := json.NewDecoder(req.Body).Decode(&options); err != nil {
			handleError(rw, err)
			return
		}
```

Reply:
x3

I see lots of very similar blocks for decoding Options from req.Body, do you think there's a way to reduce that?

I wager there also be a refactor opportunity with `handleError` where we could call another function that actually returns an error and just `handleError` once.